src/navigation/actions.js

```diff
@@ -209,12 +217,11 @@ export const createStocktake = ({ currentUser, stocktakeName, program, itemIds }
     else if (itemIds) stocktake.setItemsByID(UIDatabase, itemIds);
   });
 
-  const stackReplacementAction = StackActions.replace({
-    routeName: 'stocktakeEditor',
-    params: { stocktake, title: navStrings.stocktake },
-  });
-
-  dispatch(stackReplacementAction);
+  // If creating a Stocktake with a program or with a list of itemIds, navigate to
+  // the StocktakeEditPage. Otherwise, navigate to the StocktakeManagePage.
+  return program || itemIds
+    ? dispatch(gotoStocktakeEditPage(stocktake))
+    : dispatch(gotoStocktakeManagePage(stocktakeName, stocktake));
```

Comment:
Just personal preference, but I think is nicer to put the ternary inside the dispatch (see above)?